app/src/components/map/FeaturePopup.tsx

```diff
@@ -58,47 +102,66 @@ const useStyles = makeStyles(() => ({
   }
 }));
 
-const FeaturePopup: React.FC<React.PropsWithChildren<{ submissionSpatialComponentId: number }>> = (props) => {
-  const { submissionSpatialComponentId } = props;
+const FeaturePopup: React.FC<React.PropsWithChildren<{ submissionSpatialComponentIds: number[] }>> = (props) => {
+  const { submissionSpatialComponentIds } = props;
 
   const classes = useStyles();
   const api = useApi();
+  const [currentIndex, setCurrentIndex] = useState(0);
 
   const dataLoader = useDataLoader(() => {
-    return api.search.getSpatialMetadata(submissionSpatialComponentId);
+    return api.search.getSpatialMetadata(submissionSpatialComponentIds);
   });
 
-  /**
-   * @TODO Replace this with moment/luxon date formatter
-   */
-  const formatDate = (dateString: string) => {
-    const d = new Date(dateString);
-    return d instanceof Date && !isNaN(d as unknown as number) ? d.toDateString() : dateString;
-  };
-
   dataLoader.load();
 
-  const { isLoading, data, isReady } = dataLoader;
+  const { isLoading, isReady } = dataLoader;
```

Comment:
(Not necessarily something that needs changing, but just my thinking): I'm sure I haven't been 100% consistent on this, but I tend to lean more towards not redefining data like this, as it makes it harder further down to understand where `isLoading`, etc, came from. If you see `dataLoader.isLoading` you know exactly what isLoading is based on. It can also make it easier to modify `dataLoader` later on, because its easier to track down all of the spots it impacts, since you can just search `dataLoader` and find all of its uses.

Reply:
That's a very good point.